crates/bitwarden-core/src/client/internal.rs

```diff
@@ -354,11 +371,65 @@ impl InternalClient {
     ) -> Result<(), EncryptionSettingsError> {
         EncryptionSettings::set_org_keys(org_keys, &self.key_store)
     }
+
+    /// Updates user's KDF for the master password unlock login method.
+    /// Salt and user key update is not supported yet.
+    #[cfg(feature = "internal")]
+    pub fn update_user_master_password_unlock(
+        &self,
+        master_password_unlock: MasterPasswordUnlockData,
+    ) -> Result<(), NotAuthenticatedError> {
+        let new_kdf = master_password_unlock.kdf;
+
+        let login_method = self.get_login_method().ok_or(NotAuthenticatedError)?;
+
+        let kdf = match login_method.as_ref() {
+            LoginMethod::User(UserLoginMethod::Username { kdf, .. }) => kdf,
+            LoginMethod::User(UserLoginMethod::ApiKey { kdf, .. }) => kdf,
+            #[cfg(feature = "secrets")]
+            LoginMethod::ServiceAccount(_) => return Err(NotAuthenticatedError)?,
```

Comment:
```suggestion
            LoginMethod::ServiceAccount(_) => return Err(NotAuthenticatedError),
```

I don't think returning the error needs the unwrap.

Reply:
Fixed in ef76b78e